packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -1579,6 +1607,189 @@ test.serial('updateGranuleStatusToQueued() throws error if record does not exist
   );
 });
 
+test.serial('updateGranuleStatusToQueued() does not update DynamoDB or Elasticsearch granule if writing to PostgreSQL fails', async (t) => {
+  const {
+    collectionCumulusId,
+    esGranulesClient,
+    esClient,
+    granule,
+    granuleId,
+    granuleModel,
+    granulePgModel,
+    knex,
+  } = t.context;
+
+  const testGranulePgModel = {
+    get: () => Promise.resolve(granule),
+    update: () => {
+      throw new Error('Granules Postgres error');
+    },
+  };
+
+  await writeGranuleFromApi({ ...granule }, knex, esClient, 'Create');
+  const dynamoRecord = await granuleModel.get({ granuleId });
+  const postgresRecord = await granulePgModel.get(
+    knex,
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
+  );
+  const esRecord = await esGranulesClient.get(granuleId, granule.collectionId);
+
+  t.is(dynamoRecord.status, 'completed');
+  t.is(postgresRecord.status, 'completed');
+  t.is(esRecord.status, 'completed');
+  t.truthy(dynamoRecord.execution);
+
+  await t.throwsAsync(updateGranuleStatusToQueued({
+    granule: dynamoRecord,
+    knex,
+    granulePgModel: testGranulePgModel,
+  }));
+
+  const updatedDynamoRecord = await granuleModel.get({ granuleId });
+  const updatedPostgresRecord = await granulePgModel.get(
+    knex,
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
+  );
+  const translatedPgGranule = await translatePostgresGranuleToApiGranule({
+    granulePgRecord: updatedPostgresRecord,
+    knexOrTransaction: knex,
+  });
+  const omitList = ['_id', 'execution', 'status', 'updatedAt', 'updated_at', 'files'];
+  const sortByKeys = ['bucket', 'key'];
+
+  t.not(updatedDynamoRecord.status, 'queued');
+  t.not(updatedPostgresRecord.status, 'queued');
+  t.not(esRecord.status, 'queued');
+  t.not(updatedDynamoRecord.execution, undefined);
+  // Check that granules are equal in all data stores
+  t.deepEqual(omit(dynamoRecord, omitList), omit(updatedDynamoRecord, omitList));
+  t.deepEqual(omit(postgresRecord, omitList), omit(updatedPostgresRecord, omitList));
```

Comment:
Same throughout the rest of the tests, add `t.deepEqual(omit(esRecord, omitList), omit(updatedEsRecord, omitList))`

Reply:
[`8dadc79` (#2593)](https://github.com/nasa/cumulus/pull/2593/commits/8dadc79c1fc32caadaa9e15353bb95c92fb8ecc7)